[Bug]: test_save_figure_return seems flaky

### Bug summary

When building on Fedora builders, I often get this failure, though it is not consistent and I've never had it happen locally.

### Code for reproduction

```Python
pytest tests/test_backend_qt.py -k test_save_figure_return
```

### Actual outcome

```
___________________________ test_save_figure_return ____________________________
[gw2] linux -- Python 3.14.0 /usr/bin/python3

    @pytest.mark.backend('QtAgg', skip_on_importerror=True)
    def test_save_figure_return():
        fig, ax = plt.subplots()
        ax.imshow([[1]])
        prop = "matplotlib.backends.qt_compat.QtWidgets.QFileDialog.getSaveFileName"
        with mock.patch(prop, return_value=("foobar.png", None)):
            fname = fig.canvas.manager.toolbar.save_figure()
>           os.remove("foobar.png")
E           FileNotFoundError: [Errno 2] No such file or directory: 'foobar.png'

../BUILDROOT/usr/lib64/python3.14/site-packages/matplotlib/tests/test_backend_qt.py:228: FileNotFoundError
```

### Expected outcome

Tests always pass.

### Additional information

_No response_

### Operating system

_No response_

### Matplotlib Version

3.10.6

### Matplotlib Backend

_No response_

### Python version

_No response_

### Jupyter version

_No response_

### Installation

from source (.tar.gz)

--- a/lib/matplotlib/tests/test_backend_qt.py
+++ b/lib/matplotlib/tests/test_backend_qt.py
@@ -215,14 +215,15 @@ def test_figureoptions():
 
 
 @pytest.mark.backend('QtAgg', skip_on_importerror=True)
-def test_save_figure_return():
+def test_save_figure_return(tmp_path):
     fig, ax = plt.subplots()
     ax.imshow([[1]])
+    expected = tmp_path / "foobar.png"
     prop = "matplotlib.backends.qt_compat.QtWidgets.QFileDialog.getSaveFileName"
-    with mock.patch(prop, return_value=("foobar.png", None)):
+    with mock.patch(prop, return_value=(str(expected), None)):
         fname = fig.canvas.manager.toolbar.save_figure()
-        os.remove("foobar.png")
-        assert fname == "foobar.png"
+        assert fname == str(expected)
+        assert expected.exists()
     with mock.patch(prop, return_value=(None, None)):
         fname = fig.canvas.manager.toolbar.save_figure()
         assert fname is None

